packages/common/src/lib/workspace/shared/workspace.ts

```diff
@@ -87,6 +87,8 @@ export class Workspace<E extends object = object> {
   get active(): boolean { return this.active$.value; }
   readonly active$: BehaviorSubject<boolean> = new BehaviorSubject(false);
 
+  toolToActivate$: BehaviorSubject<{ toolbox: string; options: {[key: string]: any} }> = new BehaviorSubject(undefined);
```

Comment:
I'd like to see the PR in igo2 before accepting this. I'm wondering if it would be better to pass the toolbox to the workspace and let the action activate the right tool.

Reply:
you can check the workspace branch on igo2

https://github.com/infra-geo-ouverte/igo2/tree/workspace

Reply:
BTW, it's not a clean branch, it's a working branch! I will make cleaner pr soon.

Reply:
finally. Try it with https://github.com/infra-geo-ouverte/igo2/pull/471

Reply:
Did you take a look?

Reply:
What this tries to achieve is a bit tricky and there is no perfect solution. I understand why you did it this way. However, I think one these two approcaches would be worth testing: 1)

Reply:
We could also consider moving the wfs-actions.service and wms-actions.service to the integration package or igo2. This would allows action handler to use stuff from the states and, for example, activate the right tool from the ToolState.  In the FADQ's project, those kind of services are indeed granted access to the state.

Reply:
I will try it!

Reply:
Done. Fixed with an eventEmitter.